libfwupdplugin/fu-hidraw-device.c

```diff
@@ -353,6 +353,104 @@ fu_hidraw_device_get_feature(FuHidrawDevice *self,
 #endif
 }
 
+/**
+ * fu_hidraw_device_set_report:
+ * @self: a #FuHidrawDevice
+ * @buf: (not nullable): a buffer to use, which *must* be large enough for the request
+ * @bufsz: the size of @buf
+ * @flags: some #FuIOChannelFlags, e.g. %FU_IO_CHANNEL_FLAG_SINGLE_SHOT
+ * @error: (nullable): optional return location for an error
+ *
+ * Do a HID SetOutputReport request.
+ *
+ * Returns: %TRUE for success
+ *
+ * Since: 2.0.14
+ **/
+gboolean
+fu_hidraw_device_set_report(FuHidrawDevice *self,
+			    const guint8 *buf,
+			    gsize bufsz,
+			    FuIOChannelFlags flags,
+			    GError **error)
+{
+	g_return_val_if_fail(FU_IS_HIDRAW_DEVICE(self), FALSE);
+	g_return_val_if_fail(buf != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+#ifdef HAVE_HIDRAW_H
```

Comment:
i'm confused by the ifdefs -- I think they could just _go_, no?

Reply:
I follow what the other functions in this file do - non-linux platforms may not have the same hidraw interface.
The whole C file could be skipped during compilation but I looked at `libfwupdplugin/meson.build` and it looks like there's no other case that happens.

Reply:
Ohh right, I'll remove it

Reply:
done